### PR TITLE
Export formats as wrapper

### PR DESCRIPTION
**Pull request by JonasIsensee**
Similar to #379  except it is now based on the new `master` branch.

A net reduction in lines of code while adding functionality. ;)

This PR also fixes an issue in the latex formats where unicode was not escaped in output chunks.
(Something similar should be done for `Pandoc` since it reports
```
julia> weave("FIR_design.jmd"; doctype="pandoc2pdf", cache=:refresh, keep_unicode=false)
[WARNING] Missing character: There is no π (U+03C0) in font [lmmono12-regular]:!
[WARNING] Missing character: There is no ⋮ (U+22EE) in font [lmmono12-regular]:!
[WARNING] Missing character: There is no ⋮ (U+22EE) in font [lmmono12-regular]:!
[ Info: Weaved to /home/jonas/.julia/dev/Weave/examples/FIR_design.pdf
"/home/jonas/.julia/dev/Weave/examples/FIR_design.pdf"
```

### CODE REVIEW

**Comment by aviatesk** [comment]
@JonasIsensee thanks for your PR again, and this certainly looks so great, thanks for your suggestion ! I will make sure I will review on this until weekend.

File: src/rendering/exportformat.jl
Comment (aviatesk) [inline src/rendering/exportformat.jl:16]:
```suggestion
function Base.getproperty(sf::T, s::Symbol) where {T<:ExportFormat}
    hasfield(T, s) && return getfield(sf, s)
    return getproperty(sf.primaryformat, s)
end
function Base.setproperty!(sf::T, s::Symbol, v) where {T<:ExportFormat}
    if hasfield(T, s)
        setfield!(sf, s, v)
    else
        setproperty!(sf.primaryformat, s, v)
    end
end
function Base.hasproperty(sf::T, s::Symbol) where {T<:ExportFormat}
    hasfield(T, s) || hasproperty(sf.primaryformat, s)
end
```

File: src/rendering/htmlformats.jl
Comment (aviatesk) [inline src/rendering/htmlformats.jl:146]: why is this commented out ?

**Review by aviatesk — COMMENTED** [review]
@JonasIsensee super sorry for being late to have a review... Code looks awesome, left few comments, but I would like to merge this anyway after waiting a bit for you.

File: src/rendering/htmlformats.jl
Comment (JonasIsensee) [reply src/rendering/htmlformats.jl:146]: Very good question. I have no idea. That must have happened by accident

Comment (aviatesk) [reply src/rendering/htmlformats.jl:146]: okay, then let's keep the previous behavior for now

Comment (aviatesk) [inline src/rendering/htmlformats.jl:146]:
```suggestion
render_figures(docformat::Pandoc2HTML, chunk) = render_figures(Pandoc(), chunk)
```

**Review by aviatesk — APPROVED** [review]

**Comment by aviatesk** [comment]
@JonasIsensee well, are you still interested in this project ? I would like to give you an write access for this repository if you want :)

**Comment by JonasIsensee** [comment]
@aviatesk 

Thank you for your trust!
My work and thus the tools I need have changed a little since June and so - right now - I can't use `Weave` anymore. 
(But that could just as well change again in a few months)

I did familiarize myself with a significant part of the code base for the last few PRs so I'd be happy to help with discussion and review.

What I'm trying to say: I'll happily accept write access but in the immediate future I will not use it to actively develop.

**Pull request merged**
